bulbs/special_coverage/search.py

```
@@ -1,27 +1,110 @@
+                "included_ids": [],
+                "pinned_ids": [],
+                "groups": [],
+            })
+            for special_coverage in self._special_coverages:
+                # Access query at dict level.
+                query = getattr(special_coverage, "query", {})
+                if "query" in query:
+                    query = query.get("query")
+
+                self._query.update({
+
+                })
+                self._query["excluded_ids"] += query.get("excluded_ids", [])
+                self._query["included_ids"] += query.get("included_ids", [])
+                self._query["pinned_ids"] += query.get("pinned_ids", [])
+                self._query["groups"] += [query.get("groups", [])]
+        return self._query
+
+
+class ReadingListIterator(object):
+
+    def __init__(self, *args, **kwargs):
+        self.default_queryset = None
+        self.querysets = {}
+        self.index = 0
+
+    def __iter__(self):
+        return self
+
+    def next(self):
+        for validator, queryset in self.querysets.items():
+            if validator(self.index):
+                self.index += 1
+                return queryset.next()
+        self.index += 1
+        return self.default_queryset.next()
+
+    def __next__(self):
+        return self.next()
+
+    def register_queryset(self, queryset, validator=None, default=False):
+        """
+        Add a given queryset to the iterator with custom logic for iteration.
+
+        :param queryset: List of objects included in the reading list.
+        :param validator: Custom logic to determine a queryset's position in a reading_list.
+            Validators must accept an index as an argument and return a truthy value.
+        :param default: Sets the given queryset as the primary queryset when no validator applies.
+        """
+        if default or self.default_queryset is None:
+            self.default_queryset = queryset
+        if validator:
```

Should provide error/feedback if "no validator and not default", else it will silently ignore the queryset